pkg/cloudprovider/identity.go

```diff
@@ -35,7 +35,7 @@ func filterUserIdentity(idType *compute.ResourceIdentityType, idList *[]string,
 	case compute.ResourceIdentityTypeUserAssigned,
 		compute.ResourceIdentityTypeSystemAssignedUserAssigned:
 	default:
-		return errNotFound
+		return nil
```

Comment:
We should check for TypeNone and return nil from there, but keep the default as is.